NewProjectCI/src/main/kotlin/Main.kt

```diff
@@ -0,0 +1,87 @@
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.check
+import com.github.ajalt.clikt.parameters.options.flag
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.prompt
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.runBlocking
+import me.tongfei.progressbar.ProgressBar
+import utils.Logger.Companion.logger
+import java.lang.Thread.*
+
+class NewProjectCI : CliktCommand() {
+
+    private val packageNameRegex = "^[a-z][a-z0-9_]*(\\.[a-z0-9_]+)+[0-9a-z_]\$".toRegex()
+    val packageName: String by option(help="new package name (i.e. com.example.package)")
+        .prompt("Package name")
+        .check("(i.e. com.example.package)") { it.matches(packageNameRegex) }
+
+    private val titleRegex = "\\w+".toRegex()
+    val title: String by option(help="new app title (i.e. MyApp)")
+        .prompt("App name")
+        .check("(i.e. MyApp)") { it.matches(titleRegex) }
+
+    private val verbose by option().flag()
+
+    override fun run() = runBlocking {
+        logger.enabled = verbose
+
+        ProgressBar("New Project", 100).use { pb ->
+            pb.extraMessage = "Coping template"
+            sleep(100)
```

Comment:
Another magic number that pops up regularly, could be a constant if it's meant to be the same. And are we worried the steps are too fast? 😅

Reply:
yeah, if the steps are too fast the progress bar can't keep up